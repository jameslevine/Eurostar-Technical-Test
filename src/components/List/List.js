import React, { useEffect, useState } from "react";
import styled from "styled-components";
import { connect } from "react-redux";
import ListItem from "../ListItem/ListItem";
import fetch from "isomorphic-fetch";

const completedTasks = response => {
  const totalCompleted = response.filter(x => x.completed === true).length;
  return totalCompleted;
};

const reverseData = response => {
  return response.sort(function(a, b) {
    return b.id - a.id;
  });
};

const List = props => {
  const [response, setResponse] = useState([]);
  const [tasksCompleted, setTasksCompleted] = useState(null);
  useEffect(() => {
    fetch("/api/todos")
      .then(res => res.json())
      .then(res => {
        props.dispatch({
          type: "TOTAL_COUNT",
          payload: res.length
        });
        setTasksCompleted(completedTasks(res));
        reverseData(res);
        setResponse(res);
      })
      .catch(e => console.log(e));
  }, []);

  return (
    <>
      <Subheading>{props.list.total} results</Subheading>
      <Subheading>
        {tasksCompleted}/{props.list.total} completed
      </Subheading>
      <UnorderedList>
        {response.map(x => (
          <ListItem
            key={x.id}
            completed={x.completed}
            identification={x.id}
            title={x.title}
            tasksCompleted={tasksCompleted}
          />
        ))}
      </UnorderedList>
    </>
  );
};

export default connect(state => state)(List);

const Subheading = styled.h2`
  font-style: italic;
`;

const UnorderedList = styled.ul`
  margin: 0;
  padding: 0;
`;
