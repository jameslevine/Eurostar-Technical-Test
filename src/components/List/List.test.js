import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import List from "./List";
const mockStore = configureStore([]);

describe("My Connected React-Redux Component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      list: {
        total: 200
      }
    });
    component = renderer.create(
      <Provider store={store}>
        <List />
      </Provider>
    );
  });

  it("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
