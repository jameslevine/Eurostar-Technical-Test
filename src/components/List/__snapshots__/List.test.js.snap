// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
Array [
  <h2
    className="sc-EHOje cyUrgm"
  >
    200
     results
  </h2>,
  <h2
    className="sc-EHOje cyUrgm"
  >
    /
    200
     completed
  </h2>,
  <ul
    className="sc-bZQynM hSufOU"
  />,
]
`;
